src/metatrain/utils/data/readers/readers.py

```diff
@@ -288,8 +287,39 @@ def read_targets(
         target_info_dictionary[target_key] = TargetInfo(
             quantity=target["quantity"],
             unit=target["unit"],
-            per_atom=False,  # TODO: read this from the config
-            gradients=target_info_gradients,
+            layout=_empty_tensor_map_like(target_dictionary[target_key][0]),
         )
 
     return target_dictionary, target_info_dictionary
+
+
+def _empty_tensor_map_like(tensor_map: TensorMap) -> TensorMap:
+    new_keys = tensor_map.keys
+    new_blocks: List[TensorBlock] = []
+    for block in tensor_map.blocks():
+        new_block = _empty_tensor_block_like(block)
+        new_blocks.append(new_block)
+    return TensorMap(keys=new_keys, blocks=new_blocks)
+
+
+def _empty_tensor_block_like(tensor_block: TensorBlock) -> TensorBlock:
+    new_block = TensorBlock(
+        values=torch.empty(
+            (0,) + tensor_block.values.shape[1:],
+            dtype=torch.float64,  # metatensor can't serialize otherwise
+            device=tensor_block.values.device,
+        ),
+        samples=Labels(
+            names=tensor_block.samples.names,
+            values=torch.empty(
+                (0, tensor_block.samples.values.shape[1]),
+                dtype=tensor_block.samples.values.dtype,
+                device=tensor_block.samples.values.device,
+            ),
+        ),
+        components=tensor_block.components,
+        properties=tensor_block.properties,
+    )
+    for gradient_name, gradient in tensor_block.gradients():
+        new_block.add_gradient(gradient_name, _empty_tensor_block_like(gradient))
+    return new_block
```

Comment:
Could you open an issue to add something like this to the operations?

Reply:
These will already be gone in #376. I can still open the issue if you want

Reply:
actually, we already have it ^_^: https://docs.metatensor.org/latest/operations/reference/creation/empty_like.html